### ISSUE TITLE
Update the `transformers` integration

### ISSUE THREAD

**Issue by rlouf** [dottxt-ai/outlines#1291]
We need to update the `transformers` integration to support the new user interface.

**Comment by sky-2002** [comment]
hey @rlouf , I see this already assigned. But curious to know, what exactly needs to be changed here? I referred to the recent `llamacpp` PR in `v1.0` to get some idea, is it to separate out the inference params for different models?

**Comment by rlouf** [comment]
Yes we're separating inference arguments to simplify our life, we are also relying on their inference loop. That way integrations will be more robust and we will be able to focus on what we do best: structured generation.
